src/bin/cobalt/new.rs

```diff
@@ -409,8 +419,26 @@ pub fn publish_document(file: &path::Path) -> Result<()> {
     let doc =
         cobalt_model::DocumentBuilder::<cobalt_model::FrontmatterBuilder>::new(front, content);
     let doc = doc.to_string();
-
     cobalt_model::files::write_document_file(doc, file)?;
 
+    if config.posts.add_date_prefix {
+        // avoid prepend to existing date prefix
+        let (_, file_name) = cobalt_model::parse_file_stem(
+            file.file_stem()
+                .map(|os| os.to_string_lossy().into_owned())
+                .unwrap_or_else(|| "".to_owned()),
```

Comment:
So `parse_file_stem` takes an owned value because in its current use case, it *might* be able to reuse the allocation.

In this case, we allocate and throw it away.

On the other hand, this case isn't as performance sensitive.  So maybe we leave `parse_file_stem` alone?

Reply:
Where do you see the reuse of the allocation?

I'd like to avoid the `into_owned` if possible but if it's too much of a bother, as you said, it's only done in `publish`, so no big deal.

Reply:
> Where do you see the reuse of the allocation?

This is the current use:
https://github.com/cobalt-org/cobalt.rs/blob/6502731d043d414aa9fdb41a128088898b387cfe/src/cobalt_model/frontmatter.rs#L374

We allocate a String when extracting the `file_stem`.  If there is no prepended date, then `parse_file_stem` returns the original allocation.

Hmm, `file_stem` is then just used to create a slug, if needed, and that will create a new allocation.  I guess we can switch it to only doing borrows and would be decreasing the number of allocation (by 1 in the prepended case)